src/reducers/index.js

```diff
@@ -144,16 +144,29 @@ export function isFetchingPipelineResources(state) {
 
 export function getPipelineRuns(
   state,
-  { namespace = getSelectedNamespace(state) } = {}
+  { filters, namespace = getSelectedNamespace(state) } = {}
 ) {
-  return pipelineRunsSelectors.getPipelineRuns(state.pipelineRuns, namespace);
+  const runs = pipelineRunsSelectors.getPipelineRuns(
+    state.pipelineRuns,
+    namespace
+  );
+  return runs.filter(pipelineRun => {
+    return filters.every(filter => {
+      const [filterKey, filterValue] = filter.split(':');
+      if (filterKey && filterValue) {
+        const value = pipelineRun.metadata.labels[filterKey];
+        return value === filterValue;
+      }
+      return false;
```

Comment:
My earlier comment was not addressed.

Reply:
We did reduce the amount of code by working from what you suggested though believe that the logic flow is easier to read this way with some separation of the checks

Reply:
Totally disagree, if branches and multiple returns are definitely more complicated to read than a simple right to left boolean, it even looks like a regular sentence :sweat_smile: `return filterKey && filterValue && pipelineRun.metadata.labels[filterKey] == filterValue`

Reply:
fair enough I see what you're saying